File: halcyon-web/src/main/resources/static/js/service/function/sysFunctionService.js
/**
 * author:Marshal
 * description:service层,与数据库交互
 */
app.service("sysFunctionService",function($http){
    //base crud
    this.query=function(pageNum,pageSize,condition){
        return $http.post("../sys/function/query?pageNum="+pageNum+"&pageSize="+pageSize,condition);
    }

    this.save=function (sysFunction) {
        return $http.post("../sys/function/save",sysFunction);
    }

    this.delete=function (selectedIds) {
        return $http.get("../sys/function/delete?selectedIds="+selectedIds);
    }

    this.queryById=function (id) {
        return $http.get("../sys/function/queryById?id="+id);
    }

    this.getFunctionOptions=function () {
        return $http.get("../sys/function/getFunctionOptions");
    }
});

File: halcyon-system/src/main/java/com/marshal/halcyon/system/service/SysFunctionService.java
package com.marshal.halcyon.system.service;


import com.marshal.halcyon.system.entity.SysFunction;

import java.util.List;
import java.util.Map;

public interface SysFunctionService {
    List<SysFunction> selectFunctions(SysFunction condition, int pageNum, int pageSize);

    SysFunction selectByPrimaryKey(Long id);

    void save(SysFunction sysFunction);

    void delete(Long[] idList);

    List<Map> getFunctionOptions();

    List<SysFunction> selectTopFunctions();

    List<SysFunction> selectChildFunctions(Long functionId);
}
